### PR TITLE
Add CodeQL3000 to aspnet/buildtools pipeline

### PR DESCRIPTION
**Pull request by wtgodbe**
Runs once per week on Mondays

https://dev.azure.com/dnceng/internal/_build/results?buildId=2031067&view=results

### CODE REVIEW

File: .azure/pipelines/jobs/default-build.yml [hunk 1]
```diff
@@ -90,6 +91,17 @@ jobs:
...
       _SignType: ''
+    ${{ if eq(parameters.runCodeQL3000, 'true') }}:
+      # Do not let CodeQL3000 Extension gate scan frequency.
+      Codeql.Cadence: 0
+      # Enable CodeQL3000 unconditionally so it may be run on any branch.
+      Codeql.Enabled: true
+      # Ignore the small amount of infrastructure Python code in this repo.
+      Codeql.Language: cpp,csharp,java,javascript
+      Codeql.ExcludePathPatterns: submodules
+      Codeql.SourceRoot: src
```
Comment (wtgodbe) [inline .azure/pipelines/jobs/default-build.yml:102]: Change this

File: .azure/pipelines/ci.yml [hunk 2]
```diff
@@ -19,6 +40,8 @@ jobs:
   parameters:
     # Ensures the alignment of branch name and deployment params
     buildArgs: '/warnaserror:BUILD1001'
+    ${{ if or(eq(variables['Build.Reason'], 'Schedule'), and(eq(variables['Build.Reason'], 'Manual'), eq(parameters.runCodeQL3000, 'true'))) }}:
```
Comment (dougbu) [inline .azure/pipelines/ci.yml:43]: Narrow to run only in _internal_ (scheduled or manual w/ clicked checkbox) builds

File: .azure/pipelines/ci.yml [hunk 3]
```diff
@@ -11,6 +11,27 @@ pr:
...
     - '*'
 
+schedules:
+- cron: 0 9 * * 1
+  displayName: "Run CodeQL3000 weekly, Monday at 2:00 AM PDT"
+  branches:
+    include:
+    - release/2.1
+    - release/6.0
+    - release/7.0
```
Comment (dougbu) [inline .azure/pipelines/ci.yml:21]: Remove mention of these branches

File: .azure/pipelines/jobs/default-build.yml [hunk 4]
```diff
@@ -90,6 +91,17 @@ jobs:
...
     ${{ if or(ne(parameters.codeSign, 'true'), ne(variables['System.TeamProject'], 'internal'), eq(variables['Build.Reason'], 'PullRequest')) }}:
       _SignType: ''
+    ${{ if eq(parameters.runCodeQL3000, 'true') }}:
+      # Do not let CodeQL3000 Extension gate scan frequency.
+      Codeql.Cadence: 0
+      # Enable CodeQL3000 unconditionally so it may be run on any branch.
+      Codeql.Enabled: true
+      # Ignore the small amount of infrastructure Python code in this repo.
+      Codeql.Language: cpp,csharp,java,javascript
+      Codeql.ExcludePathPatterns: submodules
```
Comment (dougbu) [inline .azure/pipelines/jobs/default-build.yml:101]: These lines shouldn't be needed in this repo

File: .azure/pipelines/ci.yml [hunk 5]
```diff
@@ -19,6 +40,8 @@ jobs:
   parameters:
     # Ensures the alignment of branch name and deployment params
     buildArgs: '/warnaserror:BUILD1001'
+    ${{ if or(eq(variables['Build.Reason'], 'Schedule'), and(eq(variables['Build.Reason'], 'Manual'), eq(parameters.runCodeQL3000, 'true'))) }}:
+      runCodeQL3000: true
     afterBuild:
```
Comment (dougbu) [inline .azure/pipelines/ci.yml:45]: Doubt we want this `task` when doing the `CodeQL3000` work. Suggest putting it under `${{ else }}:`

Comment (wtgodbe) [reply .azure/pipelines/ci.yml:45]: This overlapping w/ the param name for the `task` was unintentional, I meant for it just to control some things in default-build. renamed.

**Comment by dougbu** [comment]
based on resolved comments, looks like this PR is missing a commit you have locally

File: .azure/pipelines/ci.yml [hunk 6]
```diff
@@ -19,6 +38,8 @@ jobs:
   parameters:
     # Ensures the alignment of branch name and deployment params
     buildArgs: '/warnaserror:BUILD1001'
+    ${{ if or(eq(variables['Build.Reason'], 'Schedule'), and(eq(variables['System.TeamProject'], 'internal'), eq(variables['Build.Reason'], 'Manual'), eq(parameters.runCodeQL3000, 'true'))) }}:
```
Comment (dougbu) [inline .azure/pipelines/ci.yml:41]: Note the schedule will run in public too. This condition doesn't handle that correctly. We want to run `CodeQL3000` internally only

File: .azure/pipelines/ci.yml (hunk 5 above)
Comment (dougbu) [reply .azure/pipelines/ci.yml:43]: My point was about uploading the KoreBuild artifact. Don't do that in a `CodeQL3000` job.

File: .azure/pipelines/ci.yml [hunk 7]
```diff
@@ -19,6 +38,8 @@ jobs:
...
+      runCodeQL3000Tasks: true
```
Comment (dougbu) [inline .azure/pipelines/ci.yml:42]: Name change is inconsistent. `runCodeQL3000Tasks` here and in default-build.yml but `runCodeQL3000` in project-ci.yml (the layer in between). Suggest going w/ `runCodeQL3000` everywhere.

File: .azure/pipelines/jobs/project-ci.yml [hunk 8]
```diff
@@ -13,37 +13,61 @@
...
-    beforeBuild: ${{ parameters.beforeBuild }}
-    afterBuild: ${{ parameters.afterBuild }}
-    variables: ${{ parameters.variables }}
+- ${{ if and(ne(variables['System.TeamProject'], 'public'), eq(parameters.runCodeQL3000, 'true')) }}:
+  - template: default-build.yml
+    parameters:
+      agentOs: Windows
+      matrix: ${{ parameters.matrix }}
+      buildArgs: ${{ parameters.buildArgs }}
+      beforeBuild:
```
Comment (dougbu) [inline .azure/pipelines/jobs/project-ci.yml:34]: Reorder the `beforeBuild` steps and add the usual `condition` to the `script` step

File: .azure/pipelines/jobs/project-ci.yml [hunk 9]
```diff
@@ -13,37 +13,61 @@
...
+  - template: default-build.yml
+    parameters:
+      agentOs: Windows
+      matrix: ${{ parameters.matrix }}
+      buildArgs: ${{ parameters.buildArgs }}
+      beforeBuild: ${{ parameters.beforeBuild }}
+      afterBuild: ${{ parameters.afterBuild }}
+      codeSign: ${{ parameters.codeSign }}
+      variables: ${{ parameters.variables }}
+      runCodeQL3000: false
```
Comment (dougbu) [inline .azure/pipelines/jobs/project-ci.yml:55]: default-build.yml doesn't have a parameter w/ this name. even if it did, no reason to set parameters to their defaults

**Review by dougbu — APPROVED** [review]

**Pull request merged**